lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -384,6 +396,8 @@ impl<'s> SegmentHolder {
     where
         F: FnMut(PointIdType, &mut RwLockWriteGuard<dyn SegmentEntry>) -> OperationResult<bool>,
     {
+        let _update_guard = self.update_tracker.update();
```

Comment:
I think we can remove this update guard.

This internally uses `apply_points` to do the actual update operation, which by itself has this guard as well.

Reply:
True, but explicit guard sets the flag a little bit earlier.